debezium-core/src/main/java/io/debezium/metrics/Metrics.java

```diff
@@ -77,14 +86,26 @@ public final void unregister(Logger logger) {
         }
     }
 
+    protected ObjectName metricName(String connectorType, String connectorName, String contextName) {
+        return metricName(connectorType, connectorName, contextName, new HashMap<>());
+    }
+
     /**
      * Create a JMX metric name for the given metric.
      * @param contextName the name of the context
      * @return the JMX metric name
-     * @throws MalformedObjectNameException if the name is invalid
      */
-    public ObjectName metricName(String connectorType, String connectorName, String contextName) {
-        final String metricName = "debezium." + connectorType.toLowerCase() + ":type=connector-metrics,context=" + contextName + ",server=" + connectorName;
+    protected ObjectName metricName(String connectorType, String connectorName, String contextName,
+                                    Map<String, String> labels) {
+        Map<String, String> allLabels = new HashMap<>();
+        allLabels.put("context", contextName);
+        allLabels.put("server", connectorName);
+        allLabels.putAll(labels);
+
+        final String metricName = "debezium." + connectorType.toLowerCase() + ":type=connector-metrics,"
+                + allLabels.entrySet().stream()
```

Comment:
the order is not guaranteed

Reply:
I did that in the beginning but the chunks of the bean name are sorted internally.